File: app/src/main/java/ru/practicum/android/diploma/team/ui/TestActivity.java
package ru.practicum.android.diploma.team.ui;


import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

import ru.practicum.android.diploma.R;

public class TestActivity extends AppCompatActivity implements SwipeStack.SwipeStackListener, View.OnClickListener {

    private Button mButtonLeft, mButtonRight;
    private FloatingActionButton mFab;

    private ArrayList<String> mData;
    private SwipeStack mSwipeStack;
    private SwipeStackAdapter mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mSwipeStack = findViewById(R.id.swipeStack);
        mFab = findViewById(R.id.fabAdd);

        mFab.setOnClickListener(this);

        mData = new ArrayList<>();
        mAdapter = new SwipeStackAdapter(mData);
        mSwipeStack.setAdapter(mAdapter);
        mSwipeStack.setListener(this);

        fillWithTestData();
    }

    private void fillWithTestData() {
        mData.add(getString(R.string.lead));
        mData.add(getString(R.string.oldwise));
        mData.add(getString(R.string.flash));
        mData.add(getString(R.string.meister));
        mData.add(getString(R.string.bitwise));
    }

    @Override
    public void onClick(View v) {
        if (v.equals(mButtonLeft)) {
            mSwipeStack.swipeTopViewToLeft();
        } else if (v.equals(mButtonRight)) {
            mSwipeStack.swipeTopViewToRight();
        } else if (v.equals(mFab)) {
            mData.add("getString(R.string.dummy_fab)");
            mAdapter.notifyDataSetChanged();
        }
    }

//    @Override
//    public boolean onCreateOptionsMenu(Menu menu) {
//        MenuInflater inflater = getMenuInflater();
//        inflater.inflate(R.menu.main, menu);
//        return true;
//    }

    //@Override
//    public boolean onOptionsItemSelected(MenuItem item) {
//
//        switch (item.getItemId()) {
//            case R.id.menuRes:
//                mSwipeStack.resetStack();
//                Snackbar.make(mFab, "R.string.stack_reset", Snackbar.LENGTH_SHORT).show();
//                return true;
//            case R.id.menuGit:
//                Snackbar.make(mFab, "R.string.stack_reset", Snackbar.LENGTH_SHORT).show();
//                return true;
//            default:
//                Snackbar.make(mFab, "R.string.stack_reset", Snackbar.LENGTH_SHORT).show();
//        }
//
//        return super.onOptionsItemSelected(item);
//    }

    @Override
    public void onViewSwipedToRight(int position) {
        String swipedElement = mAdapter.getItem(position);
        Toast.makeText(this, "Like",
                Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onViewSwipedToLeft(int position) {
        String swipedElement = mAdapter.getItem(position);
        Toast.makeText(this, "Like",
                Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onStackEmpty() {
        Toast.makeText(this, "The end", Toast.LENGTH_SHORT).show();
    }

    public class SwipeStackAdapter extends BaseAdapter {

        int[] drawableIds = {
                R.drawable.rectangle1,
                R.drawable.rectangle2,
                R.drawable.rectangle3,
                R.drawable.rectangle4,
                R.drawable.rectangle5,
                R.drawable.rectangle6
        };
        Random random = new Random();
        int randomIndex = random.nextInt(drawableIds.length);
        int randomDrawableId = drawableIds[randomIndex];


        private List<String> mData;

        public SwipeStackAdapter(List<String> data) {
            this.mData = data;
        }

        @Override
        public int getCount() {
            return mData.size();
        }

        @Override
        public String getItem(int position) {
            return mData.get(position);
        }

        @Override
        public long getItemId(int position) {
            return position;
        }

        @Override
        public View getView(final int position, View convertView, ViewGroup parent) {
            if (convertView == null) {
                convertView = getLayoutInflater().inflate(R.layout.card, parent, false);
            }

            TextView textViewCard = (TextView) convertView.findViewById(R.id.textViewCard);
            var x = mData.get(position);
            textViewCard.setText(mData.get(position));
            textViewCard.setBackgroundResource(R.drawable.rectangle1);

            return convertView;
        }
    }
}

File: app/src/main/java/ru/practicum/android/diploma/team/ui/TestActivity.kt
package ru.practicum.android.diploma.team.ui

import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.floatingactionbutton.FloatingActionButton
import ru.practicum.android.diploma.R

class TestActivity : AppCompatActivity(), SwipeStack.SwipeStackListener, View.OnClickListener {
    private var mButtonLeft: Button? = null
    private var mButtonRight: Button? = null
    private var mFab: FloatingActionButton? = null
    private var mData: ArrayList<String>? = null
    private var mSwipeStack: SwipeStack? = null
    private var mAdapter: SwipeStackAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mSwipeStack = findViewById<SwipeStack>(R.id.swipeStack)
        mButtonLeft = findViewById<Button>(R.id.buttonSwipeLeft)
        mButtonRight = findViewById<Button>(R.id.buttonSwipeRight)
        mFab = findViewById<FloatingActionButton>(R.id.fabAdd)
        mButtonLeft!!.setOnClickListener(this)
        mButtonRight!!.setOnClickListener(this)
        //mFab.setOnClickListener { }
        mData = ArrayList()
        mAdapter = SwipeStackAdapter(mData!!)
        mSwipeStack!!.adapter = mAdapter
        mSwipeStack!!.setListener(this)
        fillWithTestData()
    }

    private fun fillWithTestData() {
        for (x in 0..4) {
            mData!!.add("getString(R.string.dummy_text)" + " " + (x + 1))
        }
    }

    override fun onClick(v: View) {
        if (v == mButtonLeft) {
            mSwipeStack!!.swipeTopViewToLeft()
        } else if (v == mButtonRight) {
            mSwipeStack!!.swipeTopViewToRight()
        } else if (v == mFab) {
            mData!!.add("getString(R.string.dummy_fab)")
            mAdapter!!.notifyDataSetChanged()
        }
    }

//    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
//        val inflater: MenuInflater = getMenuInflater()
//        inflater.inflate(R.menu.main, menu)
//        return true
//    }

//    override fun onOptionsItemSelected(item: MenuItem): Boolean {
//        when (item.itemId) {
//            R.id.menuReset -> {
//                mSwipeStack!!.resetStack()
//                Snackbar.make(mFab, R.string.stack_reset, Snackbar.LENGTH_SHORT).show()
//                return true
//            }
//
//            R.id.menuGitHub -> {
//                val browserIntent = Intent(
//                    Intent.ACTION_VIEW, Uri.parse("https://github.com/flschweiger/SwipeStack")
//                )
//                startActivity(browserIntent)
//                return true
//            }
//        }
//        return super.onOptionsItemSelected(item)
//    }

    override fun onViewSwipedToRight(position: Int) {
        val swipedElement = mAdapter!!.getItem(position)
        Toast.makeText(
            this, "getString(R.string.view_swiped_right, swipedElement)",
            Toast.LENGTH_SHORT
        ).show()
    }

    override fun onViewSwipedToLeft(position: Int) {
        val swipedElement = mAdapter!!.getItem(position)
        Toast.makeText(
            this, "getString(R.string.view_swiped_left, swipedElement)",
            Toast.LENGTH_SHORT
        ).show()
    }

    override fun onStackEmpty() {
        Toast.makeText(this, "R.string.stack_empty", Toast.LENGTH_SHORT).show()
    }

    inner class SwipeStackAdapter(private val mData: List<String>) :
        BaseAdapter() {
        override fun getCount(): Int {
            return mData.size
        }

        override fun getItem(position: Int): String {
            return mData[position]
        }

        override fun getItemId(position: Int): Long {
            return position.toLong()
        }

        override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
            var convertView: View? = convertView
            if (convertView == null) {
                convertView = getLayoutInflater().inflate(R.layout.card, parent, false)
            }
            val textViewCard = convertView!!.findViewById<View>(R.id.textViewCard) as TextView
            textViewCard.text = mData[position]
            return convertView!!
        }
    }
}